lib/MediaConnection/RTCPeerMediaConnection.js

```diff
@@ -0,0 +1,158 @@
+const EventEmitter = require('events').EventEmitter;
+
+module.exports = class RTCPeerMediaConnection extends EventEmitter {
+    get LOCAL_DESCRIPTION_INDEX() { return 2 };
+
+    constructor(pcConfig, iceConnectionStateCallback)
+    {
+        super();
+
+        this.rtcPeerConnection = new RTCPeerConnection(pcConfig);
+        this.rtcPeerConnection.addEventListener('iceconnectionstatechange', () => {
+            iceConnectionStateCallback(this.rtcPeerConnection.iceConnectionState)
+        });
+    }
+
+
+    createSdpOffer(options) {
+        return this.rtcPeerConnection.createOffer(options).then(offer =>
+            Promise.all([
+                this.setLocalSdp(offer),
+                this._waitForIceReadiness(options),
+                Promise.resolve(this.getLocalSdp())])
+        ).then( result => result[this.LOCAL_DESCRIPTION_INDEX])
```

Comment:
Can we change this by:

```js
return this.rtcPeerConnection.createOffer(options)
  .then(offer => this.setLocalSdp(offer))
  .then(this._waitForIceCandidates);
```

This must happen one after the other. With `Promise.all()` is gives the impression that things happen concurrently.